### DISCUSSION TITLE
Audio not playing in browser

### DISCUSSION THREAD

**Discussion by Maddin-M** [korlibs/korge#1984] [Q&A] [answered]
How to reproduce:

1. Checkout Korge Hello World
2. Add a sound or music to the resources directory
3. Add `resourcesVfs["your-music.mp3"].readMusic().play()` to `main.kt`
4. Run `runJvm` gradle command - the sound will be playing
5. Run `runJs` gradle command - the sound will *not* be playing

(I have tested this on WIndows 11 using Firefox 115 and Microsoft Edge 115)

**Comment by ygdrasil-io** [comment]
@Maddin-M Can you please provide a minimal sample that reproduces the issue?

**Comment by Maddin-M** [comment]
@ygdrasil-io https://github.com/Maddin-M/korge-hello-world/tree/sound-problem

**Comment by soywiz** [comment, marked as answer]
Browser an explicit user interaction before allowing to play sounds. That's a limitation on browsers to avoid ADs or videos to autoplay with sound.
Once you click on the page, the sound should be start playing. If your game is for the browser you can add a scene with a button like "play". Once the click is performed sound is initialized.
